.clangd

```diff
@@ -1,2 +1,32 @@
 CompileFlags:
-  CompilationDatabase: build/       # Search build/ directory for compile_commands.json
+  # Search build/ directory for compile_commands.json
+  CompilationDatabase: build
+Diagnostics:
+  # Additional checks that are useful but too noisy to be included in .clang-tidy
+  ClangTidy:
+    Add:
+      - google-*
+      - misc-*
+      - performance-*
+      - readability-*
+    Remove:
+      - misc-no-recursion
+      - misc-unused-parameters
+      - readability-function-*
+      - readability-identifier-length
+      - readability-magic-numbers
+      - readability-redundant-function-ptr-dereference
```

Comment:
I'm not against this per se, but I can imagine it could annoy other developers that this takes priority over their own configurations. We'll need to be cautious not to add too many things at once I think. I'm not saying "remove this", but just want other people's opinions on this.

Reply:
Actually, maybe we should separate this and move it to a separate PR? It's not really related to clang-tidy so we don't need to think about this at the moment.

Reply:
this is in a ClangTidy section of .clangd. Does it affect LSP results if I'm using clangd? 

In any case, if this matches our guidelines then it makes sense to have the settings defined in our tree.

It looks like some noisy things are being removed here, which is good.

Reply:
> Does it affect LSP results if I'm using clangd?

Yes.

OK, we can try it out 👍